src/utils/train.py

```diff
@@ -188,17 +194,32 @@ def evaluate_model(
     ground_truth = torch.cat(ground_truth).numpy()
     predictions = torch.cat(predictions).numpy()
 
-    average = "binary" if num_classes == 2 else "macro"
+    if no_average:
+        average = None
+    else:
+        average = "binary" if num_classes == 2 else "macro"
     precision, recall, f1_score, _ = precision_recall_fscore_support(ground_truth, predictions, average=average)
     accuracy = accuracy_score(ground_truth, predictions)
     result = {"accuracy": accuracy, "precision": precision, "recall": recall, "f1": f1_score}
 
     if group != "":
         result = {f"{group}/{k}": v for k, v in result.items()}
 
-    if log:
-        wandb.log(result)
+    if no_average:
+        columns = ["class_name"] + [k for k, v in result.items() if not k.endswith("accuracy")]
+        data = []
+        for i in range(num_classes):
+            data.append([i] + [round(result[column][i], 3) for column in columns[1:]])
+        table = wandb.Table(data=data, columns=columns)
+        log_string = ",\n ".join(f"{k}: {v}" for k, v in result.items())
+    else:
         log_string = ", ".join(f"{k}: {round(v, 3)}" for k, v in result.items())
-        logger.info(log_string)
+
+    if log:
```

Comment:
This if can be combined with the previous one. Because you need `log_string` and `table` only when logging.